File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        f = -1 # initialize first to -1
        l = -1 # initialize last to -1

        # iteragte through array.
        for i in range(len(nums)):
            if nums[i] == target:
                if f == -1:
                    # first occurance 
                    f = i 
                
                # update last occurence
                l = i  

        # return res
        return [f, l] 

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int f = -1;
        int l = -1;

        for(int i=0; i<nums.size(); i++) {
            if(nums[i] == target) {
                if(f == -1) {
                    f = i;
                }
                l = i; 
            }
        }

        return {f , l}; 
    }
};

// time complexity:- O(n)
// space complexigty:- O(1)
